src/core/ExtensionOverrides.ts

```diff
@@ -0,0 +1,40 @@
+import { File } from './File';
+import * as fs from 'fs';
+import * as path from 'path';
+
+export class ExtensionOverrides {
+  public overrides: string[];
+
+  constructor(overrides: string[]) {
+    this.overrides = [];
+    overrides.forEach((override) => this.add(override));
+  }
+
+  private isValid(override: string) {
+    return (typeof override === 'string' && override.indexOf('.') === 0);
+  }
+
+  public add(override: string) {
+    if (this.isValid(override)) {
+      this.overrides.push(override);
+    }
+  }
+
+  public setOverrideFileInfo(file: File): string {
+    if (this.overrides.length === 0) {
+      return null;
```

Comment:
nitpick: better to return `undefined` (nothing)

Reply:
Also let's add a condition if it belongs to the actual project (vendors ignored)
`file.belongsToProject()`